purchase_blanket_order/models/purchase_order.py

```diff
@@ -0,0 +1,156 @@
+        return filters
+
+    def _get_eligible_bo_lines(self):
+        base_qty = self.product_uom._compute_quantity(
+            self.product_qty, self.product_id.uom_id)
+        filters = self._get_eligible_bo_lines_domain(base_qty)
+        return self.env['purchase.blanket.order.line'].search(filters)
+
+    @api.multi
+    def get_assigned_bo_line(self):
+        self.ensure_one()
+        eligible_bo_lines = self._get_eligible_bo_lines()
+        if eligible_bo_lines:
+            if not self.blanket_order_line or self.blanket_order_line \
+                    not in eligible_bo_lines:
+                self.blanket_order_line = \
+                    self._get_assigned_bo_line(eligible_bo_lines)
+        else:
+            self.blanket_order_line = False
+        self.onchange_blanket_order_line()
+        return {'domain': {'blanket_order_line': [
+            ('id', 'in', eligible_bo_lines.ids)]}}
+
+    @api.onchange('product_id', 'partner_id')
+    def onchange_product_id(self):
+        res = super(PurchaseOrderLine, self).onchange_product_id()
+        # If product has changed remove the relation with blanket order line
+        if self.product_id:
+            return self.get_assigned_bo_line()
+        return res
+
+    @api.onchange('product_qty', 'product_uom')
+    def _onchange_quantity(self):
+        res = super(PurchaseOrderLine, self)._onchange_quantity()
+        if self.product_id and not self.env.context.get(
+                'skip_blanket_find', False):
+            return self.get_assigned_bo_line()
+        return res
+
+    @api.onchange('blanket_order_line')
+    def onchange_blanket_order_line(self):
+        bol = self.blanket_order_line
+        if bol:
+            self.product_id = bol.product_id
+            if bol.date_schedule:
+                self.date_planned = bol.date_schedule
+            if bol.product_uom != self.product_uom:
+                price_unit = bol.product_uom._compute_price(
+                    bol.price_unit, self.product_uom)
+            else:
+                price_unit = bol.price_unit
+            self.price_unit = price_unit
+            if bol.taxes_id:
+                self.taxes_id = bol.taxes_id
+        else:
+            self._compute_tax_id()
+            self.with_context(skip_blanket_find=True)._onchange_quantity()
+
+    @api.constrains('date_planned')
+    def check_date_planned(self):
+        for line in self:
+            date_planned = fields.Date.to_string(
+                fields.Date.from_string(self.date_planned))
+            if line.blanket_order_line and \
+                    line.blanket_order_line.date_schedule and \
+                    line.blanket_order_line.date_schedule != date_planned:
+                    raise ValidationError(_(
+                        'Schedule dates defined on the Purchase Order Line '
+                        'and on the Blanket Order Line do not match.'))
+
+    @api.constrains('order_id.currency_id')
```

Comment:
This should be:
```suggestion
    @api.constrains('currency_id')
```
because if not a warning is displayed:

```
WARNING:odoo.models:method purchase.order.line.check_currency: @constrains parameter 'order_id.currency_id' is not a field name
```